### DISCUSSION TITLE
file chooser not displaying

### DISCUSSION THREAD

**Discussion by mwitz8** [fltk-rs/fltk-rs#684] [Q&A] [answered]
I have a window with a menu bar with `file<new`. I want clicking `new` to initialize a file chooser dialog, so I (try to) initialize it in the callback of `file<new`. Here's a snippet of the relevent code:
```
fn new_cb(menubar: &mut MenuBar) {
    let file_chooser = dialog::FileChooser::new(
        &"Documents",
        &"pattern",
        dialog::FileChooserType::Create,
        &"Create Project"
    );
}
let mut frame = Frame::new(0, 0, 400, 200, "");
let mut menu = MenuBar::new(0, 0, 3240, 40, "");
menu.add("File/New", enums::Shortcut::None, fltk::menu::MenuFlag::Normal, new_cb);
```
(I also don't know what to correctly input for `&"Documents"` or `&"pattern"`, but I think I can figure that out later)

The problem is, nothing seems to happen when I click `new`. Could you guys help me figure out where I'm going wrong?

**Comment by MoAlyousef** [comment, marked as answer]
The FileChooser is FLTK's own file dialog, not a wrapper around a native file dialog. It has it's own api.
Basically you'll have to call show then block:
```rust
fn new_cb(menubar: &mut menu::MenuBar) {
    let mut file_chooser = dialog::FileChooser::new(
        "./src", // a relative or absolute path to the directory you're opening
        "*.{rs,toml}", // a filter for the default shown file types
        dialog::FileChooserType::Create,
        "Create Project"
    );
    file_chooser.show();
    while file_chooser.shown() {
        app::wait();
    }
}
```

**Reply by MoAlyousef** [reply]
To use a native dialog use FileDialog like in the editor example.

**Reply by mwitz8** [reply]
thanks a ton
